nomad/structs/config/reporting.go

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package config
+
+import "github.com/hashicorp/nomad/helper/pointer"
+
+func DefaultReporting() *Reporting {
+	return &Reporting{
+		&LicenseConfig{},
+	}
+}
+
+type LicenseConfig struct {
```

Comment:
```suggestion
type LicenseReportingConfig struct {
```

`LicenseConfig` seems a bit too generic and may clash if we ever want to add license configuration values.